data_steward/analytics/cdr_ops/controlled_tier_qc/check_controlled_tier_part2.py

```diff
@@ -469,48 +471,60 @@ def my_sql(table_name, column_name):
         },
         ignore_index=True)
 
-# # query 7  All participants have basics,done
+# # query 7  All participants have basics data in the rdr dataset
```

Comment:
I have a question about the purpose of this QC.

Is this check for making sure:
a. all participants in CT having the basics records in the `CT` observation table, or
b. all participants in CT having the basics records in the `RDR` observation table?

Your update is checking `b`. But I think checking `a` is more appropriate since the basics records might be dropped along the way and they can exist in RDR but not in CT for some reason.

Let me know what you think - I might be missing some context, so I would appreciate your thoughts on this!

Reply:
That is a good question. I should have put that in the ticket. I'll do that now.
I was working off of the logic described in 'drop_participants_without_any_basics'.

```
...

As part of DC-696, several participants were found with no basics survey still persisting in the CDR
These records were removed, and DC-706 was created to remove such participants in the future. DC-2551
was created to update DropParticipantsWithoutPPI to DropParticipantsWithoutAnyBasics to
reflect what the rule is actually doing, dropping participants who have not completed any question in The Basics.
It also moves the cleaning rule to RDR_CLEANING_CLASSES list to allow curation to deliver a list of
participants expected to exist in the CT dataset when the RDR dataset is cleaned.

...

```

I saw that this CR was moved in order to enable creation of the CT expected list. To me this means that the participants who have the Basics at the RDR stage should be expected in the CDR.

Do you agree with my thought process?

Reply:
Hi Brenda, thanks for the background. I understand the context now. I have two follow-up comments.

1. Your rebase might have wiped out the changes to the raw RDR QC and the clean RDR QC. Please check the files changed in this PR again to see if it's intended.
2. Regarding my previous question, if the intention is to check `b`, I think we do not need this check in the CT QC notebook. All the participants are already confirmed to have the basics in the RDR data stage, and we are not adding any new participants in the downstream data stages. So, the updated query in the CT QC notebook will always return PASS because all the participants are guaranteed to have the basics in the RDR data stage. In the meanwhile, if the intention is to check `a`, it can return `FAILED` because the basics record might be dropped along the way.

Reply:
1. The expectations for the raw notebook are changed. The check was removed in another PR. I'll update the ticket soon.  I did make updates to both the clean rdr notebook and the CT notebook with the last commit. Thank you for the extra questions and reviews.
2. After our team discussion (see eng sync notes 'Participants with/without the Basics') we will keep the check but add to the description. I'll also make note of this in the ticket.